spec/instant.html

```diff
@@ -526,16 +526,50 @@ <h1>ToTemporalInstant ( _item_ )</h1>
       </emu-alg>
     </emu-clause>
 
-    <emu-clause id="sec-temporal-parsetemporalinstant" aoid="ParseTemporalInstant">
-      <h1>ParseTemporalInstant ( _isoString_ )</h1>
+    <emu-clause id="sec-temporal-parsetemporalinstant" type="abstract operation">
+      <h1>ParseTemporalInstant (
+        _isoString_: a String
+      ): either a normal completion containing a BigInt, or a throw completion</h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>It parses an ISO 8601 string and returns a BigInt representing a number of nanoseconds since the epoch that is represented by that string.</dd>
+      </dl>
       <emu-alg>
         1. Assert: Type(_isoString_) is String.
-        1. Let _result_ be ? ParseTemporalInstantString(_isoString_).
-        1. Let _offsetString_ be _result_.[[TimeZoneOffsetString]].
-        1. Assert: _offsetString_ is not *undefined*.
-        1. Let _utc_ be GetUTCEpochNanoseconds(_result_.[[Year]], _result_.[[Month]], _result_.[[Day]], _result_.[[Hour]], _result_.[[Minute]], _result_.[[Second]], _result_.[[Millisecond]], _result_.[[Microsecond]], _result_.[[Nanosecond]]).
-        1. Let _offsetNanoseconds_ be ? ParseDateTimeUTCOffset(_offsetString_).
-        1. Let _result_ be _utc_ - ℤ(_offsetNanoseconds_).
+        1. Let _parsed_ be ? ParseTemporalInstantString(_isoString_).
+        1. If _parsed_.[[Z]] is *true*, let _offsetNanoseconds_ be 0.
+        1. Else, let _offsetNanoseconds_ be ! ParseDateTimeUTCOffset(_parsed_.[[OffsetString]]).
+        1. Return ? GetEpochNanoseconds(_parsed_.[[Year]], _parsed_.[[Month]], _parsed_.[[Day]], _parsed_.[[Hour]], _parsed_.[[Minute]], _parsed_.[[Second]], _parsed_.[[Millisecond]], _parsed_.[[Microsecond]], _parsed_.[[Nanosecond]], _offsetNanoseconds_).
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal-getepochnanoseconds" type="abstract operation">
+      <h1>
+        GetEpochNanoseconds (
```

Comment:
I don't like the confusingly high degree of name similarity between GetEpochNanoseconds and GetUTCEpochNanoseconds. Instead of introducing this wrapping operation, I would propose keeping IsValidEpochNanoseconds checks at the invocation sites but extending a still-infallible GetUTCEpochNanoseconds to support an _offsetNanoseconds_ parameter such that e.g. InterpretISODateTimeOffset step 4 only changes from the main branch like
```diff
         1. If _offsetBehaviour_ is ~exact~ or _offsetOption_ is *"use"*, then
-          1. Let _epochNanoseconds_ be GetUTCEpochNanoseconds(_year_, _month_, _day_, _hour_, _minute_, _second_, _millisecond_, _microsecond_, _nanosecond_).
-          1. Set _epochNanoseconds_ to _epochNanoseconds_ - ℤ(_offsetNanoseconds_).
+          1. Let _epochNanoseconds_ be GetUTCEpochNanoseconds(_year_, _month_, _day_, _hour_, _minute_, _second_, _millisecond_, _microsecond_, _nanosecond_, _offsetNanoseconds_).
           1. If ! IsValidEpochNanoseconds(_epochNanoseconds_) is *false*, throw a *RangeError* exception.
           1. Return _epochNanoseconds_.
```
(but note also that IsValidEpochNanoseconds is itself infallible and therefore should not be preceded by `!`)

Reply:
OK, done!

> I would propose keeping IsValidEpochNanoseconds checks at the invocation sites but extending a still-infallible GetUTCEpochNanoseconds to support an _offsetNanoseconds_ parameter

Makes sense, this is now in the latest push. 



> (but note also that IsValidEpochNanoseconds is itself infallible and therefore should not be preceded by `!`)

This mistake was being made widely, so I added a new commit for the other call sites.

Reply:
Logistically, I'd have had a slight preference for not pulling in and editing more upstream ECMA-262 operations; GetUTCEpochNanoseconds had a single purpose and now it's slightly more complicated with an optional parameter. But I don't feel strongly about it.